packages/opentelemetry-resources/src/platform/node/detect-resources.ts

```diff
@@ -52,6 +54,59 @@ export const detectResources = async (
   );
 };
 
+/**
+ * Runs all resource detectors synchronously, merging their results. Any asynchronous
+ * attributes will be merged together in-order after they resolve.
+ *
+ * @param config Configuration for resource detection
+ */
+export const detectResourcesSync = (
+  config: ResourceDetectionConfig = {}
+): Resource => {
+  const internalConfig: ResourceDetectionConfig = Object.assign(config);
+
+  const resources: Resource[] = (internalConfig.detectors ?? []).map(d => {
+    try {
+      const resourceOrPromise = d.detect(internalConfig);
+      let resource: Resource;
+      if (resourceOrPromise instanceof Promise) {
+        const createPromise = async () => {
+          const resolved = await resourceOrPromise;
+          await resolved.waitForAsyncAttributes();
+          return resolved.attributes;
+        };
+        resource = new Resource({}, createPromise());
+      } else {
+        resource = resourceOrPromise;
+      }
+
+      resource
+        .waitForAsyncAttributes()
+        .then(() => {
+          diag.debug(`${d.constructor.name} found resource.`, resource);
+        })
+        .catch(e => {
+          diag.debug(`${d.constructor.name} failed: ${e.message}`);
+        });
+
+      return resource;
+    } catch (e) {
+      diag.debug(`${d.constructor.name} failed: ${e.message}`);
+      return Resource.empty();
+    }
+  });
+
+  const mergedResources = resources.reduce(
+    (acc, resource) => acc.merge(resource),
+    Resource.empty()
+  );
+  void mergedResources.waitForAsyncAttributes().then(() => {
+    // Future check if verbose logging is enabled issue #1903
+    logResources(resources);
```

Comment:
@open-telemetry/javascript-maintainers we have a whole node and browser implementation which are completely identical except node does this log. It really doesn't seem worth duplicating everything for just this 1 line difference. What would you think about merging the implementations and logging in both cases?

Reply:
I agree it's a good idea to merge those two.

Reply:
Should I go ahead with merging them?

Reply:
In this PR, I think merging `detectResourcesSync` is necessary to avoid creating more duplication.

Reply:
Done!

Reply:
The reason for having the two implementations was to avoid using` import * as util from 'util';` in the browser since it's not supported [(see this comment)](https://github.com/open-telemetry/opentelemetry-js/pull/3460#discussion_r1060704163)
If there's no alternative to this I will have to undo this change